### PR TITLE
[UPD] feat(setup script): additional CLI command

### PR DESCRIPTION
**Pull request by zannkukai**
* Adds a DSpace CLI command to register specific schemas and metadata fields instead of a custom python script.
* Adds a Dspace CLI command to register permission on a collection.
* Allows new metadata fields for DublinCore metadata schema.
* Removes python project dependencies.
* Allows option to clean Solr core during setup.

### CODE REVIEW

File: dspace-uclouvain/src/main/java/org/dspace/administer/CollectionPermissionManagement.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+import org.dspace.eperson.service.GroupService;
+import org.dspace.exceptions.NotUniqueResultException;
+import org.dspace.utils.DSpace;
+import org.dspace.workflow.WorkflowService;
+import org.dspace.workflow.factory.WorkflowServiceFactory;
+
+import java.sql.SQLException;
+
+/**
+ * A command-line tool for to manage permissions about a DSpace collection. It allows to
```
Comment (Pourbaix) [inline dspace-uclouvain/src/main/java/org/dspace/administer/CollectionPermissionManagement.java]: A command-line tool **to** manage

File: dspace-uclouvain/src/main/java/org/dspace/administer/CollectionPermissionManagement.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+ * add/remove permission for a roleType for a ePerson group. To allow an ePerson some
+ * management option on this group, this ePerson must be member of the previous-named group.
```
Comment (Pourbaix) [inline dspace-uclouvain/src/main/java/org/dspace/administer/CollectionPermissionManagement.java]: To **give** an ePerson some management rights on this group

File: dspace-uclouvain/src/main/java/org/dspace/administer/CollectionPermissionManagement.java [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    protected CollectionService collectionService;
+    protected WorkflowService workflowService;
+    protected SearchService searchService;
+
+
+    // CONSTRUCTOR & MAIN ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    /**
+     * For invoking via the command line.
+     *
+     * @param argv: the command line arguments given
```
Comment (Pourbaix) [inline dspace-uclouvain/src/main/java/org/dspace/administer/CollectionPermissionManagement.java:96]: the **given** command line arguments

File: scripts/setup.sh [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+#     - Check required external system command
+#   1. Check that the backend container is running
+#   2. Reinitialize the app
+#     - Clearing the database and migrating it
+#     - Reset data from Solr
+#   3. Create users:
+#     - Admin user to execute future commands
+#     - Create additional users
+#   4. Initialize collections and communities
+#   6. Execute the Python script to make the changes that are not possible via the CLI
```
Comment (Pourbaix) [inline scripts/setup.sh]: Step 5 ?

File: scripts/setup.sh [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+#
+#   X. Register project specific metadata schemas & fields.
+#   X. Assign users into groups
+
+# Define constants used during script execution
+readonly BACKEND="dspace"
```
Comment (Pourbaix) [inline scripts/setup.sh:44]: Could be great to be able to run this script for locally installed TOMCAT :smile:

File: scripts/setup.sh [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+	if ! command -v "$cmd" &> /dev/null
+  then
+      error_msg+exit  "⛔ \`${RED}${cmd}${NC}\` could not be found. This command is required to complete the script."
+      exit 1
+  fi
+done
+
+# Create the log file if it does not exists
+[ ! -d "${WORKING_PATH}/log" ] && mkdir "${WORKING_PATH}/log"
+echo -e "📃 Logs for this execution can be find in: ${CYAN}${LOG_PATH}${NC}"
```
Comment (Pourbaix) [inline scripts/setup.sh]: Logs for this execution can be **found** in:

File: dspace-uclouvain/src/main/java/org/dspace/administer/GroupManagement.java [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+        CommandLine cl = gm.validateCLIArgument(argv);
+        gm.manageGroup(
+                cl.getOptionValue("a"),
+                cl.getOptionValue("n"),
+                cl.hasOption("f")
+        );
+    }
+
+    /**
+     * constructor, which just creates and object with a ready context
```
Comment (Pourbaix) [inline dspace-uclouvain/src/main/java/org/dspace/administer/GroupManagement.java]: just creates **an** object

File: dspace-uclouvain/src/main/java/org/dspace/administer/GroupManagement.java [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+        return GroupManagement.USAGE_DESCRIPTION;
+    }
+
+    // PRIVATE FUNCTIONS ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    /**
+     * Perform a management group action
+     *
+     * @param action: the action to execute (see constants of this class to find available actions)
+     * @param name: the group name.
+     * @param force: is the operation should be forced
```
Comment (Pourbaix) [inline dspace-uclouvain/src/main/java/org/dspace/administer/GroupManagement.java]: should the operation be forced / if the operation should be forced

File: dspace-uclouvain/src/main/java/org/dspace/administer/UserGroupManagement.java [hunk 9]
```diff
@@ -187,16 +169,25 @@ private EPerson findUser(String userID) throws IllegalStateException {
     }
 
     /**
-     * Find a group based on its name
+     * Get a group based on its name. If this group doesn't yet exist, it could be
+     * created.
      *
      * @param groupName: the group name
+     * @param createGroup: If group doesn't yet exist, should it create ?
```
Comment (Pourbaix) [inline dspace-uclouvain/src/main/java/org/dspace/administer/UserGroupManagement.java]: If group doesn't yet exist, **should it be created** ?

**Review by Pourbaix — COMMENTED** [review]
Just little English tweaks here and there ( :smiling_imp:  ), but the rest seems good to me :smiley:.
EDIT: I tried to build a Docker image (to test the script) and it didn't worked out with the `docker build --no-cache -f Dockerfile . ` command on this branch. I did it on another branch (uclouvain-dev) and it worked. Strange...

File: scripts/setup.sh (hunk 5 above)
Comment (zannkukai) [reply scripts/setup.sh:44]: in next PR ; new feature

**Review by Pourbaix — APPROVED** [review]

**Pull request merged**